# Simpson's Paradox
Use `admission_data.csv` for this exercise.

In [104]:
# Load and view first few lines of dataset
import pandas as pd
df = pd.read_csv('admission_data.csv')

### Proportion and admission rate for each gender

In [89]:
# get variables
total_females = df.query('gender == "female"').gender.count()
total_students = df.student_id.count()

# Proportion of students that are female
female_proportion = (total_females/total_students) * 100
female_percent = round(female_proportion,0)
print(female_percent, '% of students are females')

51.0 % of students are females


In [90]:
# get variables
total_males = df.query('gender == "male"').gender.count()
total_students = df.student_id.count()

# Proportion of students that are male
male_proportion = (total_males/total_students) * 100
male_percent = round(male_proportion,0)
print(male_percent, '% of students are males')

49.0 % of students are males


In [91]:
# Admission rate for females
total_females = df.query('gender == "female"').gender.count()
adm_females = df.query('gender == "female" & admitted == True').admitted.count()

female_admission = (adm_females/total_females) * 100
female_admission_percent = round(female_admission,0)
print(female_admission_percent, '% admitted are females')

29.0 % admitted are females


In [92]:
# Admission rate for males
total_males = df.query('gender == "male"').gender.count()
adm_males = df.query('gender == "male" & admitted == True').admitted.count()

male_admission = (adm_males/total_males) * 100
male_admission_percent = round(male_admission,0)
print(male_admission_percent, '% admitted are males')

49.0 % admitted are males


### Proportion and admission rate for physics majors of each gender

In [49]:
# What proportion of female students are majoring in physics?
fem_physics = df.query(' gender == "female" & major == "Physics" ').student_id.count()
fem_physics_percent = (fem_physics/total_females) * 100
fem_physics_percent = round(fem_physics_percent,0)
print(fem_physics_percent, '% of females are majoring in Physics')

12.0 % of females are majoring in Physics


In [50]:
# What proportion of male students are majoring in physics?
male_physics = df.query(' gender == "male" & major == "Physics" ').student_id.count()
male_physics_percent = (male_physics/total_males) * 100
male_physics_percent = round(male_physics_percent,0)
print(male_physics_percent, '% of males are majoring in Physics')

93.0 % of males are majoring in Physics


In [65]:
# Admission rate for female physics majors
fem_admitted_physics = df.query(' gender == "female" & \
                                  admitted == True &  \
                                  major == "Physics" ').student_id.count()
fem_admitted_percent = (fem_admitted_physics/fem_physics) * 100
fem_admitted_percent = round(fem_admitted_percent,0)
print(fem_admitted_percent,'% of females admitted are majoring in Physics')

74.0 % of females admitted are majoring in Physics


In [66]:
# Admission rate for male physics majors
male_admitted_physics = df.query(' gender == "male" & \
                                  admitted == True &  \
                                  major == "Physics" ').student_id.count()
male_admitted_percent = (male_admitted_physics/male_physics) * 100
male_admitted_percent = round(male_admitted_percent,0)
print(male_admitted_percent,'% of males admitted are majoring in Physics')

52.0 % of males admitted are majoring in Physics


### Proportion and admission rate for chemistry majors of each gender

In [101]:
# What proportion of female students are majoring in chemistry?
fem_chem = df.query(' gender == "female" & major == "Chemistry" ').student_id.count()
fem_chem_percent = (fem_chem/total_females) * 100
fem_chem_percent = round(fem_chem_percent,0)
print(fem_chem_percent, '% of females are majoring in Chemistry')

88.0 % of females are majoring in Chemistry


In [100]:
# What proportion of male students are majoring in chemistry?
male_chem = df.query(' gender == "male" & major == "Chemistry" ').student_id.count()
male_chem_percent = (male_chem/total_males) * 100
male_chem_percent = round(male_chem_percent,0)
print(male_chem_percent, '% of males are majoring in Chemistry')

7.0 % of males are majoring in Chemistry


In [76]:
# Admission rate for female chemistry majors
fem_admitted_chem = df.query(' gender == "female" & \
                               admitted == True &  \
                               major == "Chemistry" ').student_id.count()
fem_admitted_percent = (fem_admitted_chem/fem_chem) * 100
fem_admitted_percent = round(fem_admitted_percent,0)
print(fem_admitted_percent,'% of females admitted are majoring in Chemistry')

23.0 % of females admitted are majoring in Chemistry


In [80]:
# Admission rate for male chemistry majors
male_admitted_chem = df.query(' gender == "male" & \
                                  admitted == True &  \
                                  major == "Chemistry" ').student_id.count()
male_admitted_percent = (male_admitted_chem/male_chem) * 100
male_admitted_percent = round(male_admitted_percent,0)
print(male_admitted_percent,'% of males admitted are majoring in Chemistry')

11.0 % of males admitted are majoring in Chemistry


### Admission rate for each major

In [81]:
# Admission rate for physics majors
admitted_physics = df.query(' admitted == True &  \
                              major == "Physics" ').student_id.count()
admitted_percent = (admitted_physics/(fem_physics + male_physics)) * 100
admitted_percent = round(admitted_percent,0)
print(admitted_percent,'% of students admitted are majoring in Physics')

54.0 % of students admitted are majoring in Physics


In [102]:
# Admission rate for chemistry majors
admitted_chemistry = df.query(' admitted == True &  \
                              major == "Chemistry" ').student_id.count()
admitted_percent = (admitted_chemistry/(fem_chem + male_chem)) * 100
admitted_percent = round(admitted_percent,0)
print(admitted_percent,'% of students admitted are majoring in Chemistry')

22.0 % of students admitted are majoring in Chemistry


* From this dataset, **males** tend to be **admitted at a higher rate at 49%** when compared to females at 29%. 
* **Physics majors are 93% males** while only 12% for females, **however more females are admitted to major in Physics at 74%** comparing to males at 52%.
* **Chemistry majors are 88% females** and 7% are only males. For students that are **admitted to major in Chemistry, 23% are females** and 11% are males. 
* **More students are majoring in Physics at 54%** compared to Chemistry at 22%